src/components/big-interactive-pages/editor.tsx

```diff
@@ -57,29 +56,29 @@ export const saveGame = debounce(800, (persistenceState: Signal<PersistenceState
 					body: JSON.stringify({ code, gameId: game?.id, tutorialName: game?.tutorialName, tutorialIndex: game?.tutorialIndex })
 				})
 				if (!res.ok) throw new Error(`Error saving game: ${await res.text()}`)
-				isError = false
+				return true;
 			} catch (error) {
 				console.error(error)
-				isError = true
 
 				persistenceState.value = {
 					...persistenceState.value,
 					cloudSaveState: 'ERROR'
 				} as any;
+				return false;
 			}
 		}
 
-		await attemptSaveGame();
-		while (isError) {
-			await attemptSaveGame();
+		let error = !await attemptSaveGame();
+		while (error) {
```

Comment:
can't we just do: while(!await attemptSaveGame()) ?

Reply:
also - what is the consequence of this loop running for a long time?  This will not block the user's UI, will it?  Can this end up resulting in multiple error loops happening in parallel?  Sort of concerned that we might want to make this more asynchronous (a separate timer mechanism)...?

Reply:
aha i pushed some more updates already

Reply:
this has not blocked any more functionality in the editor as far as I've tested